hatch_build.py

```diff
@@ -0,0 +1,83 @@
+from hatchling.builders.hooks.plugin.interface import BuildHookInterface
+from hatchling.metadata.plugin.interface import MetadataHookInterface
+
+import tomlkit
+import pathlib
+
+
+class PinVersionMetadataHook(MetadataHookInterface):
+    """Update the dependency versions so they are pinned metadata of the wheel artefact produced by a wheel."""
+    def update(self, metadata):
+        change_count = 0
+        # Overlay the pinned dependencies, replacing dependencies if they already exist.
+        for pinned_dep in self.config["config"]["pinned_dependencies"]:
+            pinned_dep_name = pinned_dep.split(" ")[0]
+            for dep in metadata["dependencies"]:
+                dep_name = dep.split(" ")[0]
+                # If the dependency is already in the list, replace it with the pinned version.
+                if dep_name == pinned_dep_name:
+                    index = metadata["dependencies"].index(dep)
+                    metadata["dependencies"][index] = pinned_dep
+                    change_count += 1
+                    break
+        print(f"Updated {change_count} dependencies to pinned versions in hatch's metadata "
```

Comment:
Thank you for these diagnostic prints, they are super-comforting to understanding what's actually going on when the code is running.

Reply:
It seemed wise to provide at least some output to confirm it was in deed doing the dependency overrides. Given how `hatch` doesn't give a lot of information about what it's doing under the hood when running.